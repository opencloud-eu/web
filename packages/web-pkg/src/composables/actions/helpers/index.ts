export * from './useIsSearchActive'
export * from './useFileActionsDeleteResources'
export * from './useIsFilesAppActive'
export * from './useIsAppActive'
