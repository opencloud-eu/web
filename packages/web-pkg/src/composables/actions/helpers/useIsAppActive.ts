import { computed, unref } from 'vue'
import { useRoute } from '../../router'

export const useIsAppActive = () => {
  const currentRoute = useRoute()
  return computed(() => unref(currentRoute)?.query?.contextRouteName)
}
