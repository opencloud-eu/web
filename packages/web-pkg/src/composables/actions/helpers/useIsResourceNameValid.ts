import { Resource } from '@opencloud-eu/web-client'
import { useGettext } from 'vue3-gettext'
import { RESOURCE_MAX_CHARACTER_LENGTH } from '../../../constants'
import { useResourcesStore } from '../../piniaStores'

export const useIsResourceNameValid = () => {
  const { $gettext } = useGettext()
  const resourcesStore = useResourcesStore()

  const isFileNameValid = (
    resource: Resource,
    newName: string,
    parentResources: Resource[] = undefined
  ): { isValid: boolean; error?: string } => {
    if (!newName) {
      return { isValid: false, error: $gettext('The name cannot be empty') }
    }

    if (/[/]/.test(newName)) {
      return { isValid: false, error: $gettext('The name cannot contain "/"') }
    }

    if (newName === '.') {
      return { isValid: false, error: $gettext('The name cannot be equal to "."') }
    }

    if (newName === '..') {
      return { isValid: false, error: $gettext('The name cannot be equal to ".."') }
    }

    if (newName.trim() !== newName) {
      return { isValid: false, error: $gettext('The name cannot start or end with whitespace') }
    }

    if (newName.length > RESOURCE_MAX_CHARACTER_LENGTH) {
      return {
        isValid: false,
        error: $gettext('The name cannot be longer than %{length} characters', {
          length: RESOURCE_MAX_CHARACTER_LENGTH.toString()
        })
      }
    }

    const newPath =
      resource.path.substring(0, resource.path.length - resource.name.length) + newName
    const exists = (parentResources || resourcesStore.resources).some(
      (file) => file.path === newPath && file.name === newName
    )
    if (exists) {
      const translated = $gettext('The name »%{name}« is already taken')
      return { isValid: false, error: $gettext(translated, { name: newName }, true) }
    }

    return { isValid: true, error: undefined }
  }

  const isSpaceNameValid = (newName: string): { isValid: boolean; error?: string } => {
    if (newName.trim() === '') {
      return {
        isValid: false,
        error: $gettext('The Space name cannot be empty')
      }
    }

    if (newName.length > RESOURCE_MAX_CHARACTER_LENGTH) {
      return {
        isValid: false,
        error: $gettext('The Space name cannot be longer than %{length} characters', {
          length: RESOURCE_MAX_CHARACTER_LENGTH.toString()
        })
      }
    }

    if (newName.trim() !== newName) {
      return {
        isValid: false,
        error: $gettext('The Space name cannot start or end with whitespace')
      }
    }

    if (/[/\\.:?*"><|]/.test(newName)) {
      return {
        isValid: false,
        error: $gettext(
          'The Space name cannot contain the following characters: / \\\\ . : ? * " > < |\''
        )
      }
    }

    return { isValid: true, error: undefined }
  }

  return {
    isSpaceNameValid,
    isFileNameValid
  }
}
