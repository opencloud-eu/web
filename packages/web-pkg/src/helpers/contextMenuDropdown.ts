import { OcDrop } from '@opencloud-eu/design-system/components'
import { ComponentPublicInstance } from 'vue'

export type ContextMenuBtnClickEventData = {
  event: MouseEvent | KeyboardEvent
  dropdown: ComponentPublicInstance<typeof OcDrop>
}

const isKeyboardEvent = (event: Event): event is KeyboardEvent => {
  return (event as any).clientY === 0
}

export const displayPositionedDropdown = (
  dropdown: ComponentPublicInstance<typeof OcDrop>,
  event: MouseEvent | KeyboardEvent,
  contextMenuButton: ComponentPublicInstance<unknown>
) => {
  const contextMenuButtonPos = contextMenuButton.$el.getBoundingClientRect()

  const yValue = isKeyboardEvent(event)
    ? (event.target as HTMLElement)?.getBoundingClientRect().top || 0
    : event.clientY

  dropdown.setProps({
    getReferenceClientRect: () => ({
      width: 0,
      height: 0,
      top: yValue,
      bottom: yValue,
      /**
       * If event type is 'contextmenu' the trigger was a right click on the table row,
       * so we render the dropdown at the position of the mouse pointer.
       * Otherwise we render the dropdown at the position of the three-dot-menu
       */
      left:
        event.type === 'contextmenu' && !isKeyboardEvent(event)
          ? event.clientX
          : contextMenuButtonPos.x,
      right:
        event.type === 'contextmenu' && !isKeyboardEvent(event)
          ? event.clientX
          : contextMenuButtonPos.x
    })
  })

  dropdown.show()
}
