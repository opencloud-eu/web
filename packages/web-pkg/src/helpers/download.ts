export const triggerDownloadWithFilename = (url: string, name = '') => {
  const a = document.createElement('a')
  a.style.display = 'none'
  document.body.appendChild(a)
  a.href = url
  // use download attribute to set desired file name
  a.setAttribute('download', name)
  // trigger the download by simulating click
  a.click()
  // cleanup
  document.body.removeChild(a)
}
