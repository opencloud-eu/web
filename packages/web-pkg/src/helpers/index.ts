export * from './cache'
export * from './folderLink'
export * from './resource'
export * from './router'
export * from './share'
export * from './spaces'
export * from './ui'

export * from './breadcrumbs'
export * from './clipboardActions'
export * from './contextMenuDropdown'
export * from './datetime'
export * from './download'
export * from './filesize'
export * from './fuse'
export * from './locale'
export * from './path'
export * from './statusIndicators'
export * from './store'
export * from './binary'
export * from './platform'
