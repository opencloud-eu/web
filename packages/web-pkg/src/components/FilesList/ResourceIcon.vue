<template>
  <oc-icon
    :key="`resource-icon-${icon.name}`"
    :name="icon.name"
    :color="icon.color"
    :size="size"
    :class="['oc-resource-icon', iconTypeClass]"
  />
</template>

<script lang="ts">
import { computed, defineComponent, inject, PropType, unref } from 'vue'
import {
  isPersonalSpaceResource,
  isProjectSpaceResource,
  Resource,
  SpaceResource
} from '@opencloud-eu/web-client'
import { AVAILABLE_SIZES, SizeType } from '@opencloud-eu/design-system/helpers'
import {
  createDefaultFileIconMapping,
  IconType,
  ResourceIconMapping,
  resourceIconMappingInjectionKey
} from '../../helpers/resource/icon'

const defaultFolderIcon: IconType = {
  name: 'resource-type-folder',
  color: 'var(--oc-color-icon-folder)'
}

const defaultPersonalSpaceIcon: IconType = {
  name: 'resource-type-folder',
  color: 'var(--oc-role-secondary)'
}

const defaultSpaceIcon: IconType = {
  name: 'layout-grid',
  color: 'var(--oc-role-secondary)'
}

const defaultSpaceIconDisabled: IconType = {
  name: 'layout-grid',
  color: 'var(--oc-role-secondary)'
}

const defaultFallbackIcon: IconType = {
  name: 'resource-type-file',
  color: 'var(--oc-role-on-surface)'
}

const defaultFileIconMapping = createDefaultFileIconMapping()

export default defineComponent({
  name: 'ResourceIcon',
  props: {
    /**
     * The resource to be displayed
     */
    resource: {
      type: Object as PropType<Resource | SpaceResource>,
      required: true
    },
    /**
     * The size of the icon. Defaults to small.
     * `xsmall, small, medium, large, xlarge, xxlarge`
     */
    size: {
      type: String as PropType<SizeType>,
      default: 'large',
      validator: (value: string): boolean => {
        return AVAILABLE_SIZES.some((e) => e === value)
      }
    }
  },
  setup(props) {
    const iconMappingInjection = inject<ResourceIconMapping>(resourceIconMappingInjectionKey)

    const isFolder = computed(() => {
      // fallback is necessary since
      // sometimes resources without a type
      // but with `isFolder` are being passed
      return props.resource.type === 'folder' || props.resource.isFolder
    })

    const isSpace = computed(() => {
      return props.resource.type === 'space'
    })

    const isDisabledSpace = computed(() => {
      return isProjectSpaceResource(props.resource) && props.resource.disabled === true
    })

    const isPersonalSpace = computed(() => {
      return isPersonalSpaceResource(props.resource)
    })
    const extension = computed(() => {
      return props.resource.extension?.toLowerCase()
    })
    const mimeType = computed(() => {
      return props.resource.mimeType?.toLowerCase()
    })

    const icon = computed((): IconType => {
      if (unref(isPersonalSpace)) {
        return defaultPersonalSpaceIcon
      }
      if (unref(isDisabledSpace)) {
        return defaultSpaceIconDisabled
      }
      if (unref(isSpace)) {
        return defaultSpaceIcon
      }
      if (unref(isFolder)) {
        return defaultFolderIcon
      }

      const icon =
        defaultFileIconMapping[unref(extension)] ||
        iconMappingInjection?.mimeType[unref(mimeType)] ||
        iconMappingInjection?.extension[unref(extension)]

      return {
        ...defaultFallbackIcon,
        ...icon
      }
    })

    const iconTypeClass = computed(() => {
      if (unref(isDisabledSpace)) {
        return 'oc-resource-icon-space-disabled'
      }
      if (unref(isSpace)) {
        return 'oc-resource-icon-space'
      }
      if (unref(isFolder)) {
        return 'oc-resource-icon-folder'
      }
      return 'oc-resource-icon-file'
    })

    return {
      icon,
      iconTypeClass
    }
  }
})
</script>

<style lang="scss">
span.oc-resource-icon {
  display: inline-flex;
  align-items: center;
  vertical-align: middle;

  &-file svg {
    height: 70%;
  }
}

span.oc-resource-icon-space-disabled {
  filter: grayscale(100%);
  opacity: 80%;
}
</style>
