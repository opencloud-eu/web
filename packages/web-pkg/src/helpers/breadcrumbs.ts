import { eventBus } from '../services'
import { RouteLocation } from 'vue-router'
import omit from 'lodash-es/omit'
import { BreadcrumbItem } from '@opencloud-eu/design-system/helpers'
import { v4 as uuidV4 } from 'uuid'

export const breadcrumbsFromPath = (
  currentRoute: RouteLocation,
  resourcePath: string
): BreadcrumbItem[] => {
  const pathSplit = (p = '') => p.split('/').filter(Boolean)
  const current = pathSplit(currentRoute.path)
  const resource = pathSplit(resourcePath)

  return resource.map(
    (text, i) =>
      ({
        id: uuidV4(),
        allowContextActions: true,
        text,
        to: {
          path: '/' + [...current].splice(0, current.length - resource.length + i + 1).join('/'),
          query: omit(currentRoute.query, 'fileId', 'page') // TODO: we need the correct fileId in the query. until we have that we must omit it because otherwise we would correct the path to the one of the (wrong) fileId.
        },
        isStaticNav: false
      }) as BreadcrumbItem
  )
}

export const concatBreadcrumbs = (...items: BreadcrumbItem[]): BreadcrumbItem[] => {
  const last = items.pop()

  return [
    ...items,
    {
      id: uuidV4(),
      allowContextActions: last.allowContextActions,
      text: last.text,
      onClick: () => eventBus.publish('app.files.list.load'),
      isTruncationPlaceholder: last.isTruncationPlaceholder,
      isStaticNav: last.isStaticNav
    }
  ]
}
