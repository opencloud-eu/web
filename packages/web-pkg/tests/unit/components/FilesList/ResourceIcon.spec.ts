import { shallowMount } from '@opencloud-eu/web-test-helpers'
import { AVAILABLE_SIZES } from '@opencloud-eu/design-system/helpers'
import ResourceIcon from '../../../../src/components/FilesList/ResourceIcon.vue'
import {
  ResourceIconMapping,
  resourceIconMappingInjectionKey
} from '../../../../src/helpers/resource'
import { Resource } from '@opencloud-eu/web-client'

const resourceIconMapping: ResourceIconMapping = {
  extension: {
    'not-a-real-extension': {
      name: 'resource-type-madeup-extension',
      color: 'red'
    }
  },
  mimeType: {
    'not-a-real-mimetype': {
      name: 'resource-type-file',
      color: 'var(--oc-role-on-surface)'
    }
  }
}

describe('OcResourceIcon', () => {
  ;['file', 'folder', 'space'].forEach((type) => {
    match({
      type
    })
  })

  match(
    {
      type: 'file',
      extension: 'not-a-real-extension'
    },
    'with extension "not-a-real-extension"'
  )

  match(
    {
      type: 'file',
      mimeType: 'not-a-real-mimetype'
    },
    'with mimetype "not-a-real-mimetype"'
  )
})

function match(resource: Partial<Resource>, additionalText?: string) {
  AVAILABLE_SIZES.forEach((size) => {
    it(`renders OcIcon for resource type ${resource.type}${
      additionalText ? ` ${additionalText}` : ''
    } in size ${size}`, () => {
      const { wrapper } = getWrapper({ resource, size })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
}

function getWrapper({ resource, size }: { resource: Partial<Resource>; size: string }) {
  return {
    wrapper: shallowMount(ResourceIcon, {
      global: {
        provide: {
          [resourceIconMappingInjectionKey]: resourceIconMapping
        }
      },
      props: {
        resource: resource as Resource,
        size
      }
    })
  }
}
