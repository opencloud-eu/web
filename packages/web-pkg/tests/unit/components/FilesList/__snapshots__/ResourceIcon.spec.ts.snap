// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OcResourceIcon > renders OcIcon for resource type file in size large 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="large" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file in size medium 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="medium" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file in size small 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="small" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file in size xlarge 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file in size xsmall 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xsmall" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file in size xxlarge 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xxlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file in size xxxlarge 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xxxlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with extension "not-a-real-extension" in size large 1`] = `"<oc-icon name="resource-type-madeup-extension" color="red" size="large" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with extension "not-a-real-extension" in size medium 1`] = `"<oc-icon name="resource-type-madeup-extension" color="red" size="medium" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with extension "not-a-real-extension" in size small 1`] = `"<oc-icon name="resource-type-madeup-extension" color="red" size="small" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with extension "not-a-real-extension" in size xlarge 1`] = `"<oc-icon name="resource-type-madeup-extension" color="red" size="xlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with extension "not-a-real-extension" in size xsmall 1`] = `"<oc-icon name="resource-type-madeup-extension" color="red" size="xsmall" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with extension "not-a-real-extension" in size xxlarge 1`] = `"<oc-icon name="resource-type-madeup-extension" color="red" size="xxlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with extension "not-a-real-extension" in size xxxlarge 1`] = `"<oc-icon name="resource-type-madeup-extension" color="red" size="xxxlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with mimetype "not-a-real-mimetype" in size large 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="large" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with mimetype "not-a-real-mimetype" in size medium 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="medium" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with mimetype "not-a-real-mimetype" in size small 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="small" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with mimetype "not-a-real-mimetype" in size xlarge 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with mimetype "not-a-real-mimetype" in size xsmall 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xsmall" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with mimetype "not-a-real-mimetype" in size xxlarge 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xxlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type file with mimetype "not-a-real-mimetype" in size xxxlarge 1`] = `"<oc-icon name="resource-type-file" color="var(--oc-role-on-surface)" size="xxxlarge" class="oc-resource-icon oc-resource-icon-file"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type folder in size large 1`] = `"<oc-icon name="resource-type-folder" color="var(--oc-color-icon-folder)" size="large" class="oc-resource-icon oc-resource-icon-folder"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type folder in size medium 1`] = `"<oc-icon name="resource-type-folder" color="var(--oc-color-icon-folder)" size="medium" class="oc-resource-icon oc-resource-icon-folder"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type folder in size small 1`] = `"<oc-icon name="resource-type-folder" color="var(--oc-color-icon-folder)" size="small" class="oc-resource-icon oc-resource-icon-folder"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type folder in size xlarge 1`] = `"<oc-icon name="resource-type-folder" color="var(--oc-color-icon-folder)" size="xlarge" class="oc-resource-icon oc-resource-icon-folder"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type folder in size xsmall 1`] = `"<oc-icon name="resource-type-folder" color="var(--oc-color-icon-folder)" size="xsmall" class="oc-resource-icon oc-resource-icon-folder"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type folder in size xxlarge 1`] = `"<oc-icon name="resource-type-folder" color="var(--oc-color-icon-folder)" size="xxlarge" class="oc-resource-icon oc-resource-icon-folder"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type folder in size xxxlarge 1`] = `"<oc-icon name="resource-type-folder" color="var(--oc-color-icon-folder)" size="xxxlarge" class="oc-resource-icon oc-resource-icon-folder"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type space in size large 1`] = `"<oc-icon name="layout-grid" color="var(--oc-role-secondary)" size="large" class="oc-resource-icon oc-resource-icon-space"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type space in size medium 1`] = `"<oc-icon name="layout-grid" color="var(--oc-role-secondary)" size="medium" class="oc-resource-icon oc-resource-icon-space"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type space in size small 1`] = `"<oc-icon name="layout-grid" color="var(--oc-role-secondary)" size="small" class="oc-resource-icon oc-resource-icon-space"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type space in size xlarge 1`] = `"<oc-icon name="layout-grid" color="var(--oc-role-secondary)" size="xlarge" class="oc-resource-icon oc-resource-icon-space"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type space in size xsmall 1`] = `"<oc-icon name="layout-grid" color="var(--oc-role-secondary)" size="xsmall" class="oc-resource-icon oc-resource-icon-space"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type space in size xxlarge 1`] = `"<oc-icon name="layout-grid" color="var(--oc-role-secondary)" size="xxlarge" class="oc-resource-icon oc-resource-icon-space"></oc-icon>"`;

exports[`OcResourceIcon > renders OcIcon for resource type space in size xxxlarge 1`] = `"<oc-icon name="layout-grid" color="var(--oc-role-secondary)" size="xxxlarge" class="oc-resource-icon oc-resource-icon-space"></oc-icon>"`;
